services/frontend-service/src/ui/Pages/Home/Home.test.tsx

```diff
@@ -31,21 +31,242 @@ describe('App', () => {
+
+    const data: dataT[] = [
+        {
+            name: 'gets every app - 2 results',
+            selectedTeams: ['dummy', 'foo'],
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: 'dummy',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'test',
+                },
+                example: {
+                    name: 'example',
+                    releases: [],
+                    sourceRepoUrl: 'http://example.com',
+                    team: 'test2',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expectedNumOfTeams: 2,
+        },
+        {
+            name: 'shows both applications of the selected team - 2 results',
+            selectedTeams: ['dummy'],
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: 'dummy',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'dummy',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expectedNumOfTeams: 2,
+        },
+        {
+            name: 'no teams selected (shows every application) - 4 results',
+            selectedTeams: [],
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: '',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'test',
+                },
+                example: {
+                    name: 'example',
+                    releases: [],
+                    sourceRepoUrl: 'http://example.com',
+                    team: '',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expectedNumOfTeams: 4,
+        },
+    ];
+
+    describe.each(data)(`Renders an Application Card`, (testcase) => {
```

Comment:
I would test as well the case where there is one filter for a team (for example `test`) and there is no application for that team